Engine/src/Engine/ResourceManager.cpp

```diff
@@ -89,6 +89,26 @@ namespace Engine
 		}
 	}
 
+	//Save provided json object to a provided file. Defaults file name and path to "Data/data.json"
+	//If the provided file name does not exist, it will be created. Otherwise, the existing file will be overwritten.
+	void ResourceManager::saveJsonFile(nlohmann::json data, std::string fileName, std::string path)
+	{
+		//Create a file path from the provided path, file name, and file extension.
+		std::string fileWithPath = path + fileName + "." + m_jsonFileExt;
+
+		std::ofstream fileStream(fileWithPath);	// Create a file stream and open the file
+		if (fileStream.is_open())
+		{
+			fileStream << data;					// Replace existing file content with the updated prevData
+			fileStream.close();					// Close the file stream so it doesn't hang open
+		}
+		else
+		{
+			// This should never happen but is left in for debugging purposes.
+			GE_CORE_ERROR("[ResourceManager] The file stream broke!", fileWithPath);
```

Comment:
I think this error is inaccurate. If a filestream isn't open, that just means the file couldn't be read/opened. The file stream would only break on line 102, if at all (not sure I've seen that happen though). Maybe adjust the message?

Reply:
updated it to be more accurate